src/test/java/io/vertx/test/core/FileDescriptorLeakDetectorRule.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.test.core;
+
+import com.sun.management.UnixOperatingSystemMXBean;
```

Comment:
we don't even want to import this class because it will not load in windows

Reply:
we should use the platform `MBeanServer` and invoke this as an `MBean` using its `ObjectName` and do a `getAttribute` on `OpenFileDescriptorCount`.

before that we want to check the mbean metadata to see if the info exist or nor (on windows it will nto exist) using the `MBeanInfo` returned by the `MBeanServer`.

Reply:
On windows this rule would then just do nothing.

Reply:
I'll handle this and continue the PR @zenios

Reply:
Really sorry @vietj  i got blocked by a an urgent project :(

Reply:
N/p @zenios you did most of the hard work :-)

Here is the continuation PR https://github.com/eclipse-vertx/vert.x/pull/3408